### PR TITLE
New Resource: azurerm_securitycenter_contact

### PR DESCRIPTION
**Pull request by katbyte**

### CODE REVIEW

**Comment by katbyte** [comment]
#2043 should be merged first

File: azurerm/resource_arm_securitycenter_contact.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+	return resourceArmSecurityCenterContactRead(d, meta)
+}
+
+func resourceArmSecurityCenterContactRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).securityCenterContactsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	//id is in format of `/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/securityContacts/john`
+	//parseAzureResourceID doesn't support id without a resource group
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_securitycenter_contact.go]: we should probably think about creating an overload which doesn't require it - there's a few instances of this now (key vault)

File: azurerm/resource_arm_securitycenter_contact_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+					resource.TestCheckResourceAttr(resourceName, "alert_notifications", "true"),
+					resource.TestCheckResourceAttr(resourceName, "alerts_to_admins", "true"),
+				),
+			},
+			{
+				Config: testAccAzureRMSecurityCenterContact_template("email2@example.com", "+1-555-678-6789", false, false),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSecurityCenterContactExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "email", "email2@example.com"),
+					resource.TestCheckResourceAttr(resourceName, "phone", "+1-555-678-6789"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_securitycenter_contact_test.go]: could we make this an international number to ensure that works too? `+44 1234 567890` should work as an example

File: azurerm/resource_arm_securitycenter_contact.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+//seems you can only set one contact:
+// Invalid security contact name was provided - only 'defaultX' is allowed where X is an index
+// Invalid security contact name 'default0' was provided. Expected 'default1'
+// Message="Invalid security contact name 'default2' was provided. Expected 'default1'"
+
+func resourceArmSecurityCenterContact() *schema.Resource {
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/resource_arm_securitycenter_contact.go]: can we document this resource too?

File: azurerm/resource_arm_security_center_contact.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"log"
+)
+
+//seems you can only set one contact:
+// Invalid security contact name was provided - only 'defaultX' is allowed where X is an index
+// Invalid security contact name 'default0' was provided. Expected 'default1'
+// Message="Invalid security contact name 'default2' was provided. Expected 'default1'"
+const resourceArmSecurityCenterContactName = "default1"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_security_center_contact.go]: _minor_ can we ditch `resourceArm` at the start of this

File: azurerm/resource_arm_security_center_subscription_pricing.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+	if properties := resp.PricingProperties; properties != nil {
+		d.Set("tier", properties.PricingTier)
+	}
+
+	return nil
+}
+
+func resourceArmSecurityCenterSubscriptionPricingDelete(_ *schema.ResourceData, _ interface{}) error {
+	return nil //cannot be deleted.
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_security_center_subscription_pricing.go]: IMO we should probably log this?

File: website/docs/r/security_center_contact.markdown [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+The following attributes are exported:
+
+* `id` - The Security Center Contact ID.
+
+## Import
+
+The contact can be imported using the `resource id`, e.g.
+
+```shell
+terraform import azurerm_securitycenter_contact.example /subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Security/securityContacts/default1
```
Comment (tombuildsstuff) [inline website/docs/r/security_center_contact.markdown]: `securitycenter` -> `security_center`

File: website/azurerm.erb [hunk 7]
```diff
@@ -973,6 +973,20 @@
...
+                <li<%= sidebar_current("docs-azurerm-security-center-subscription-pricing") %>>
+                  <a href="/docs/providers/azurerm/r/security_center_subscription_pricing.html">azurerm_security_center_subscription_pricing</a>
+                </li>
+              </ul>
+              <ul class="nav nav-visible">
+              <li<%= sidebar_current("docs-azurerm-security-center-contact") %>>
+                <a href="/docs/providers/azurerm/r/security_center_contact.html">azurerm_security_center_contact</a>
+                </li>
+              </ul>
+            </li>
```
Comment (tombuildsstuff) [inline website/azurerm.erb]: _minor_ it'd be good to sort these alphabetically

**Pull request merged**